client.go

```diff
@@ -92,16 +95,25 @@ func newClient(config Config) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	c := &Client{
+	c = &Client{
 		ic:                      ic,
 		lifecycleListenerMap:    map[types.UUID]int64{},
 		lifecycleListenerMapMu:  &sync.Mutex{},
 		membershipListenerMap:   map[types.UUID]int64{},
 		membershipListenerMapMu: &sync.Mutex{},
 		nearCacheMgrsMu:         &sync.RWMutex{},
-		nearCacheMgrs:           map[string]inearcache.Manager{},
+		nearCacheMgrs:           map[string]*inearcache.Manager{},
 		cfg:                     &config,
 	}
+	if c.ic.StatsService != nil {
+		c.ic.StatsService.SetNCStatsGetter(func(service string) stats.NearCacheStatsGetter {
+			ncmgr, ok := c.nearCacheMgrs[service]
+			if !ok {
+				return nil
+			}
+			return ncmgr
+		})
```

Comment:
We can also consider setting this in createComponents, in `if config.StatsEnabled {` block.